### ISSUE TITLE
"linker `x86_64-w64-mingw32.static.posix-gcc.exe` not found" error on Windows build

### ISSUE THREAD

**Issue by eitsupi** [pola-rs/r-polars#153]
It seems to be occurring after #121 was merged, but #121 seems to be unrelated to this error, so I do not know what is causing it.

```log
── R CMD build ─────────────────────────────────────────────────────────────────
* checking for file 'D:\a\r-polars\r-polars/DESCRIPTION' ... OK
* preparing 'polars':
* checking DESCRIPTION meta-information ... OK
* cleaning src
* installing the package to build vignettes
      -----------------------------------
* installing *source* package 'polars' ...
** using staged installation
** libs
rm -Rf polars.dll ./rust/target/x86_64-pc-windows-gnu/release/libpolars.a entrypoint.o
gcc  -I"C:/R/include" -DNDEBUG     -I"c:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -Wall -pedantic -fdiagnostics-color=always -c entrypoint.c -o entrypoint.o
mkdir -p ./rust/target/libgcc_mock && touch ./rust/target/libgcc_mock/libgcc_eh.a
# CARGO_LINKER is provided in Makevars.ucrt for R >= 4.2
#--target-dir ./rust/target
echo "RPOLARS_RUST_SOURCE is set to /d/a/r-polars/r-polars/src/rust"
RPOLARS_RUST_SOURCE is set to /d/a/r-polars/r-polars/src/rust
export CARGO_TARGET_X86_64_PC_WINDOWS_GNU_LINKER="x86_64-w64-mingw32.static.posix-gcc.exe" && \
	export LIBRARY_PATH="${LIBRARY_PATH};/c/Users/runneradmin/AppData/Local/Temp/RtmpaSK6DM/Rbuildb78576a20b3/polars/src/./rust/target/libgcc_mock" && \
	cargo +nightly-gnu build --target=x86_64-pc-windows-gnu --lib --release --manifest-path="/d/a/r-polars/r-polars/src/rust/Cargo.toml"
   Compiling proc-macro2 v1.0.50
   Compiling unicode-ident v1.0.6
   Compiling quote v1.0.23
error: linker `x86_64-w64-mingw32.static.posix-gcc.exe` not found
  |
  = note: program not found
error: could not compile `proc-macro2` (build script) due to previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `quote` (build script) due to previous error
make: *** [Makevars.win:23: rust/target/x86_64-pc-windows-gnu/release/libpolars.a] Error 101
ERROR: compilation failed for package 'polars'
* removing 'C:/Users/RUNNER~1/AppData/Local/Temp/RtmpaSK6DM/Rinstb781f114a3c/polars'
      -----------------------------------
ERROR: package installation failed
Error: Error in proc$get_built_file() : Build process failed
Calls: <Anonymous> ... build_package -> with_envvar -> force -> <Anonymous>
Execution halted
Error: Process completed with exit code 1.
Run ## --------------------------------------------------------------------
Show testthat output
Run actions/upload-artifact@v3
Warning: No files were found with the provided path: D:/a/r-polars/r-polars/check. No artifacts will be uploaded.
```

**Comment by sorhawell** [comment]
@eitsupi 
I believe it is the release of R 4.3
I kick myself for not running develop windows version in all checks.

I propose to remove windows from main check for now to not halt all development.

Instead I will run all windows builds in a branch a debug the problem.
I will certainly bump extendr, maybe there is already a fix.

**Comment by Ilia-Kosenkov** [comment]
`extendr` uses a different linker for the fresh version of R on Windows, the one you can find in `$env:RTOOLS43_HOME\x86_64-w64-mingw32.static.posix\bin`. Add this to `PATH` and it will work.
Now, I am not sure why it is not automatically resolved, I was under the impression that R adds correct paths to Rtools during the compilation phase. Perhaps worth an issue in `extendr/extendr`.

**Comment by sorhawell** [comment]
@Ilia-Kosenkov 

fixed in WIP PR #155 there was an r-lib/actions bug which was fixed 20 min after [reporting](https://github.com/r-lib/actions/issues/720)

**Comment by eitsupi** [comment]
I think this was fixed by #168
